torch/_tensor.py

```diff
@@ -846,6 +846,31 @@ def symeig(self, eigenvectors=False):
 
         return _symeig(self, eigenvectors=eigenvectors)
 
+    def cumsum(
+        self,
+        dim=None,
+        *,
+        dtype=None,
+        out=None,
+        axis=None,
+    ):
+        r"""
+        cumsum(dim, dtype=None) -> Tensor
+
+        See :func:`torch.cumsum`
+        """
+        if has_torch_function_unary(self):
+            return handle_torch_function(
+                Tensor.cumsum,
+                (self,),
+                self,
+                dim,
+                dtype=dtype,
+                out=out,
+                axis=axis,
+            )
+        return torch.cumsum(self, dim, dtype=dtype, out=out, axis=axis)
```

Comment:
Hmm, I guess we need to do it here too?

Reply:
Done. Also, I figured we should avoid giving both axis and dim